test_bst.py

```diff
@@ -123,3 +143,86 @@ def test_unbalanced_to_the_left(balanced):
 def test_unbalanced_to_the_right(unbalanced):
     '''Graphs unbalanced to the right should return the appropriate negative value.'''
     assert unbalanced.balance() < 0
+
+
+def test_in_order(balanced):
+    '''Test generator in_order with balanced graph.'''
+    expected = [2, 3, 4, 5, 6, 7, 10, 11]
+    for expected, actual in zip(expected, balanced.in_order()):
+        assert expected == actual
+
+
+def test_in_order_big(big):
+    '''Test generator in_order with big graph.'''
+    expected = range(1, 51)
+    for expected, actual in zip(expected, big.in_order()):
+        assert expected == actual
+
+
+def test_pre_order(balanced):
+    '''Test generator pre_order with balanced graph.'''
+    expected = [5, 4, 3, 2, 10, 7, 6, 11]
+    for expected, actual in zip(expected, balanced.pre_order()):
+        assert expected == actual
+
+
+def test_pre_order_big(big):
+    '''Test generator pre_order with balanced graph.'''
+    expected = range(50, 0, -1)
+    for expected, actual in zip(expected, big.pre_order()):
+        assert expected == actual
+
+
+def test_post_order(balanced):
+    '''Test generator post_order with balanced graph.'''
+    expected = [2, 3, 4, 6, 7, 11, 10, 5]
+    for expected, actual in zip(expected, balanced.post_order()):
+        assert expected == actual
+
+
+def test_post_order_big(big):
+    '''Test generator post_order with balanced graph.'''
+    expected = range(1, 51)
+    for expected, actual in zip(expected, big.post_order()):
+        assert expected == actual
+
+
+def test_breadth_first(balanced):
+    '''Test generator breadth_first with balanced graph.'''
+    expected = [5, 4, 10, 3, 7, 11, 2, 6]
+    for expected, actual in zip(expected, balanced.breadth_first()):
+        assert expected == actual
```

Comment:
What happens when you zip a list with 8 items in it with a generator that produces no items?

```python
>>> foo = (val for val in [])
>>> type(foo)
<type 'generator'>
>>> zip([1,2,3,4,5,6,7,8], foo)
[]
```
This test doesn't actually prove anything because your `breadth_first` traversal doesn't ever yield anything.  

Be careful about the assumptions you make when testing.  Just because tests are passing doesn't mean they aren't broken.